### DISCUSSION TITLE
Will kustomization support non-standard conditions?

### DISCUSSION THREAD

**Discussion by jack-evans** [fluxcd/flux2#2457] [Proposals]
Currently, Kustomization supports health checks for most standard k8s objects, the CRs that make up the gitops toolkit and any CRs that support kstatus computation.

The problem is that the CRs I am looking to use don't follow the kstatus "rules". For example, kyverno CRs have .Status.Ready which is a Boolean. ExternalSecrets do use Ready but neither the Reconciling nor Stalled state. The only ways to get around it seems to be not to healthcheck, to wrap in a Helm chart or use a pod/job to deploy the change. 

I guess my proposal could be 1 of a couple things:

1. Health checks support the ability to define a non standard (maybe in json path notation or Rego) to determine if a resource is ready. Perhaps take inspiration from argoCD? 

2. Currently in the HealthChecks you have to define all the resources to check, it would be great if rather you could define a set of includes or a set of excludes. Excludes would allow for a large kustomization to reduce bloat on the health checks if it contains non standard resources that cannot be checked

**Comment by kingdonb** [comment]
Both of those sound like good ideas, let's see what the consensus of the community is behind this.

It could be useful to make health checking customizable, probably there are lots of projects that do not agree with Kstatus (CAPI, ExternalSecrets, ... how many more? it seems there is at least a sizeable list) – thanks for bringing this to us. Let's think about it?
